crates/bevy_ecs/src/schedule/condition.rs

```diff
@@ -685,14 +685,14 @@ pub mod common_conditions {
     /// app.run(&mut world);
     /// assert_eq!(world.resource::<Counter>().0, 1);
     ///
-    /// *world.resource_mut::<State<GameState>>() = State(GameState::Paused);
+    /// *world.resource_mut::<State<GameState>>() = State::new(GameState::Paused);
     ///
     /// // Now that we are in `GameState::Pause`, `pause_system` will run
     /// app.run(&mut world);
     /// assert_eq!(world.resource::<Counter>().0, 0);
     /// ```
     pub fn in_state<S: States>(state: S) -> impl FnMut(Res<State<S>>) -> bool + Clone {
-        move |current_state: Res<State<S>>| current_state.0 == state
+        move |current_state: Res<State<S>>| *current_state.get() == state
```

Comment:
`States` has a `Clone` bound already, so I think it should just return by value 100% of the time.

Reply:
I'm not sure I think implicitly cloning for getting the state would be great, I'd be more comfortable with impling on just Copy + Clone ones though.

If we want to simplify this a bit we could just impl PartialEq/Eq on `State<S>` itself and it could just be
`current_state == state`

Reply:
I like the impl on State quite a bit actually: that really feels like the idiomatic approach.

Reply:
Yup way better!